rasa/core/nlg/response.py

```diff
@@ -34,7 +34,13 @@ def _matches_filled_slots(
         for constraint in constraints:
             name = constraint["name"]
             value = constraint["value"]
-            if filled_slots.get(name) != value:
+            filled_slots_value = filled_slots.get(name)
+            is_string_check = isinstance(filled_slots_value, str) and isinstance(
+                value, str
+            )
+            if is_string_check and (filled_slots_value.casefold() == value.casefold()):
+                return True
```

Comment:
```suggestion
            if isinstance(filled_slots_value, str) and isinstance(value, str):
                return filled_slots_value.casefold() == value.casefold()
```
Then we can remove the variable `is_string_check`, you can check locally to see if mypy check passes by running `make types`.

Reply:
Thank you